core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -69,16 +71,26 @@ object TopicCommand extends Logging {
       else if (opts.hasDeleteOption)
         topicService.deleteTopic(opts)
     } catch {
+      case e: ExecutionException =>
+        if (e.getCause != null)
+          printException(e.getCause)
+        else
+          printException(e)
+        exitCode = 1
       case e: Throwable =>
-        println("Error while executing topic command : " + e.getMessage)
-        error(Utils.stackTrace(e))
+        printException(e)
         exitCode = 1
     } finally {
       topicService.close()
       Exit.exit(exitCode)
     }
   }
 
+  private def printException(e: Throwable): Unit = {
+    println("Error while executing topic command : " + e.getMessage)
+    error(Utils.stackTrace(e))
```

Comment:
Couldn't we just use `error(message, e)`?

Reply:
Do you mean using `error(message, e)` to replace both `println` and `error`? I think that we are using 'println` here in order to print the message to stdout without any logger related stuff and regardless of how the logger is configured. Changing to using `error(message, e)` would break this and potentially break existing application due to introducing the logger related stuff for that message. I think that we should keep `println` here.

However, I wonder if using `error` is appropriate here as it basically prints a stacktrace for every errors. The UX does not look good. Would it make sense to use `debug` instead? The message of the exception is printed anyway and I don't think that the stacktrace provides much to regular users. WDYT?

Reply:
Sorry, I should have been more clear. I just meant, why do we call `Utils.stackTrace(e))` when `error(message, e)` gives you a stack trace for free. In terms of changing from `error` to `debug`, I think we include stack track in other commands. But each command seems to use something different. ConfigCommand, for example, doesn't print stacktrace for config exceptions. It logs both the error message and stacktrace to stderr. AclCommand prints for everything to stdout. Let's just keep `error` for now and maybe open a JIRA to improve and make tools consistent later?

Reply:
That's a very good question. I guess that we did so in order to just print out the stacktrace without the message. Otherwise, we would have the error message printed out by the `println`, followed by the same message printed out by logger, followed by the stacktrace. Having the message twice is not necessary. I had a quick look at other commands and we do so everywhere.

I will open a JIRA to make tools more consistent. Good idea.